src/components/Header/AuthenticatedHeaderOption.js

```diff
@@ -0,0 +1,464 @@
+import { actions as userActions } from "../User/redux/action";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import ExpandLessIcon from "@mui/icons-material/ExpandLess";
+import useStyles from "./styles";
+import { DropDown, MobileDropDown } from "./DropDown";
+import { breakpoints } from "../../theme/constant";
+import useMediaQuery from "@mui/material/useMediaQuery";
+
+import {
+  Box,
+  IconButton,
+  Typography,
+  Menu,
+  Avatar,
+  MenuItem,
+} from "@mui/material";
+
+const SwitchView = ({
+  role,
+  setSwitchView,
+  handleCloseSwitchView,
+  switchView,
+}) => {
+  const classes = useStyles();
+  if (role == "admin") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.PARTNERS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  if (role == "volunteer") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.CLASS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  if (role == "partner") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.PARTNERS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  return "";
+};
+
+function AuthenticatedHeaderOption({ toggleDrawer, leftDrawer }) {
+  const [partnerId, setPartnerId] = useState("");
+  const [profile, setProfile] = useState("");
+  const [anchorElUser, setAnchorElUser] = React.useState(null);
+  const [learn, setLearn] = React.useState(null);
+  const [dropDown, setDropDown] = React.useState(null);
+  const [studentView, setStudentView] = React.useState(false);
+  const [switchView, setSwitchView] = React.useState("");
+  const dispatch = useDispatch();
+  const user = useSelector(({ User }) => User);
+  const rolesList = user.data.user.rolesList;
+  const classes = useStyles();
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/users/me`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      setPartnerId(res.data.user.partner_id);
+      setProfile(res.data.user.profile_picture);
+    });
+  }, []);
```

Comment:
Redux-Saga or temporarily:
```suggestion
  useEffect(() => {
    sendToken({token: user.data.token}).then((res) => {
      setPartnerId(res.data.user.partner_id);
      setProfile(res.data.user.profile_picture);
    });
  }, []);
```